### ISSUE TITLE
Request support for Firefox Send！

### ISSUE THREAD

**Issue by 9E307** [ShareX/ShareX#2736]
It is recommended that file uploads support "FireFox Send" upload options.

**Comment by NovusTheory** [comment]
The API is unknown and the JS code behind it is obfuscated. Not possible at the moment because there is no traditional API to do it with. (Not to mention ShareX would then need to encrypt client side, which is also unknown how FireFox Send does it)

**Comment by Scrxtchy** [comment]
@NovusTheory overthinking it, I could get this done in in 5 minutes

**Comment by Scrxtchy** [comment]
Probably needs a bit more work
Using a hardcoded X-File-Metadata header, I need to be able to pass the file extension to the custom uploader which I don't believe is implemented according to the tooltips @Jaex 

otherwise, works fine
![image](https://user-images.githubusercontent.com/12771982/29754884-e6927dc2-8bd1-11e7-9a85-1961fb6881c2.png)

**Comment by NovusTheory** [comment]
Interesting, I thought there was more complication into it

**Comment by Scrxtchy** [comment]
Yeah, the only thing I really should look into is the ID variable in the json. It won't be too hard, just something I that doesn't feel comfortable when working on a tablet using edge

**Comment by Jaex** [comment]
I thought Firefox Send would require client side encryption of data, not sure how it works without.

**Comment by Scrxtchy** [comment]
Haha, it should be, guess it works without encryption

**Comment by vatterspun** [comment]
I hate to say this because I really like Mozilla and I hope the project continues, but given the program is in their "test pilot" section and even the URL has the word "experiments" in it, it would probably be unwise to spend much time on it until it's promoted to something more solid.

**Comment by Scrxtchy** [comment]
That's why it's just a custom uploader

**Comment by Jirubizu** [comment]
anyone have an updated working example now?

**Comment by NovusTheory** [comment]
@Jirubizu Something like this can't entirely be added until Mozilla creates a public API. Reverse engineering is possible but not ideal.

There is an active feature request here for it https://github.com/mozilla/send/issues/1225

**Comment by alex-che** [comment]
https://github.com/timvisee/ffsend - this guy managed to implement a command line client for Firefox Send.

**Comment by Jaex** [comment]
@alex-che feel free to pull request Firefox Send support then.

**Comment by Scrxtchy** [comment]
The appropriate source to look at would be this
https://github.com/timvisee/ffsend-api/tree/master/src

**Comment by NovusTheory** [comment]
@Jaex You should re-open this issue if you're going to actively use this one
